### PR TITLE
Optimization Handling (part 2)

### PR DESCRIPTION
**Pull request by G-Moris**
Optimization Handling (part 2), Continuation of PR - #3482

### CODE REVIEW

File: Client/multiplayer_sa/CMultiplayerSA.cpp [hunk 1]
```diff
@@ -6118,7 +6118,7 @@ bool                 CheckHasSuspensionChanged()
 
         CModelInfo* pModelInfo = pGameInterface->GetModelInfo(pVehicle->GetModelIndex());
         if (pModelInfo && (pModelInfo->IsCar() || pModelInfo->IsMonsterTruck()))
-            return pVehicle->GetHandlingData()->HasSuspensionChanged();
+            return true;
```
Comment (tederis) [inline Client/multiplayer_sa/CMultiplayerSA.cpp:6116]: Why did you replace a function call with `true`?

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 2]
```diff
@@ -186,35 +189,33 @@ CHandlingManagerSA::CHandlingManagerSA()
...
     m_HandlingNames["animGroup"] = HANDLING_ANIMGROUP;                    // works model based
-
-    iChangedVehicles = 0;
 }
 
 CHandlingManagerSA::~CHandlingManagerSA()
 {
     // Destroy all original handling entries
-    for (int i = 0; i < HT_MAX; i++)
+    for (std::uint8_t i = 0; i < HT_MAX; i++)
```
Comment (tederis) [inline Client/game_sa/CHandlingManagerSA.cpp]: Please, use `size_t`. The same applies to other places.

File: Client/game_sa/CHandlingManagerSA.cpp [hunk 3]
```diff
@@ -24,27 +24,28 @@ extern CGameSA*        pGame;
 #define Var_fTurnMassMultiplier     0x858B8C
 #define Var_fBasicDragCoeff         0x858C58
 
-DWORD CHandlingManagerSA::m_dwStore_LoadHandlingCfg = 0;
+#define DUMP_HANDLING_DATA FALSE
```
Comment (tederis) [inline Client/game_sa/CHandlingManagerSA.cpp]: There should be zero

File: Client/game_sa/CHandlingEntrySA.h [hunk 4]
```diff
@@ -95,111 +96,105 @@ class CHandlingEntrySA : public CHandlingEntry
...
-    void Recalculate(unsigned short usModel);
+    void SetHeadLight(eLightType Style) { m_Handling.ucHeadLight = Style; }
+    void SetTailLight(eLightType Style) { m_Handling.ucTailLight = Style; }
+    void SetAnimGroup(unsigned char ucGroup) { m_Handling.ucAnimGroup = ucGroup; }
 
-    tHandlingDataSA* GetInterface() { return m_pHandlingSA; };
+    void Recalculate();
 
-    void SetSuspensionChanged(bool bChanged) { m_bChanged = bChanged; };
+    tHandlingDataSA* GetInterface() { return m_pHandlingSA; }
```
Comment (tederis) [inline Client/game_sa/CHandlingEntrySA.h]: const

File: Client/game_sa/CHandlingEntrySA.h [hunk 5]
```diff
@@ -95,111 +96,105 @@ class CHandlingEntrySA : public CHandlingEntry
...
+    void SetTractionBias(float fTractionBias) { m_Handling.fTractionBias = fTractionBias; }
 
-    void SetCollisionDamageMultiplier(float fMultiplier) { m_Handling.fCollisionDamageMultiplier = fMultiplier; };
+    void SetSuspensionForceLevel(float fForce) { m_Handling.fSuspensionForceLevel = fForce; }
+    void SetSuspensionDamping(float fDamping) { m_Handling.fSuspensionDamping = fDamping; }
+    void SetSuspensionHighSpeedDamping(float fDamping) { m_Handling.fSuspensionHighSpdDamping = fDamping; }
+    void SetSuspensionUpperLimit(float fUpperLimit) { m_Handling.fSuspensionUpperLimit = fUpperLimit; }
+    void SetSuspensionLowerLimit(float fLowerLimit) { m_Handling.fSuspensionLowerLimit = fLowerLimit; }
+    void SetSuspensionFrontRearBias(float fBias) { m_Handling.fSuspensionFrontRearBias = fBias; }
+    void SetSuspensionAntiDiveMultiplier(float fAntidive) { m_Handling.fSuspensionAntiDiveMultiplier = fAntidive; }
```
Comment (tederis) [inline Client/game_sa/CHandlingEntrySA.h]: Before the PR there was CheckSuspensionChanges call within these methods. Why did you remove it?

File: Client/game_sa/CHandlingEntrySA.cpp [hunk 6]
```diff
@@ -22,91 +22,42 @@ CHandlingEntrySA::CHandlingEntrySA()
...
 }
 
 CHandlingEntrySA::~CHandlingEntrySA()
 {
-    if (m_bChanged)
-    {
-        pGame->GetHandlingManager()->RemoveChangedVehicle();
-    }
-    if (m_bDeleteInterface)
+    if (m_bDeleteInterface && m_pHandlingSA)
```
Comment (tederis) [inline Client/game_sa/CHandlingEntrySA.cpp]: Unnecessary. It's okay if you delete nullptr.

**Review by tederis — CHANGES_REQUESTED** [review]

File: Client/multiplayer_sa/CMultiplayerSA.cpp (hunk 1 above)
Comment (G-Moris) [reply Client/multiplayer_sa/CMultiplayerSA.cpp:6116]: The HasSuspensionChanged returned only true

File: Client/game_sa/CHandlingEntrySA.h (hunk 5 above)
Comment (G-Moris) [reply Client/game_sa/CHandlingEntrySA.h]: CheckSuspensionChanges is useless, you can see for yourself

File: Client/game_sa/CHandlingManagerSA.cpp (hunk 2 above)
Comment (G-Moris) [reply Client/game_sa/CHandlingManagerSA.cpp]: Good

File: Client/game_sa/CHandlingEntrySA.h (hunk 5 above)
Comment (G-Moris) [reply Client/game_sa/CHandlingEntrySA.h]: @tederis Can I end this conversation?

Comment (tederis) [reply Client/game_sa/CHandlingEntrySA.h]: Please, do not mark a conversation resolved if the actual conversastion is in process. CheckSuspensionChanges has a side effect: internally it can call SetSuspensionEnabled. So it's not useless.

Comment (G-Moris) [reply Client/game_sa/CHandlingEntrySA.h:178]: > Please, do not mark a conversation resolved if the actual conversastion is in process. CheckSuspensionChanges has a side effect: internally it can call SetSuspensionEnabled. So it's not useless.

Ok. Have you looked at SetSuspensionEnabled?

Comment (G-Moris) [reply Client/game_sa/CHandlingEntrySA.h]: Ok. Maybe I was wrong and I will return it with changes.

Comment (G-Moris) [reply Client/game_sa/CHandlingEntrySA.h]: @tederis Can I mark a conversation resolved?

Comment (Dutchman101) [reply Client/game_sa/CHandlingEntrySA.h]: Yes, it's resolved, because you did the desired thing (re-add those calls).

**Comment by Dutchman101** [comment]
The code reviews that were completed in this PR before, were all done on request (to scrutinize the PR, same as previous refactor PR by the author), so with great care. So that now that all review comments have been addressed, it should be good for merge.
